pyflakes/checker.py

```diff
@@ -1072,6 +1074,8 @@ def addBinding(self, node, value):
                 break
         existing = scope.get(value.name)
 
+        value.during_type_checking = self._in_type_checking
```

Comment:
my comment before was because attribute stuffing is a smell (and a type checker such as mypy would error accordingly on this line), this should either be:

1. added to each of the types as part of their `__init__` signature
2. stored separately alongside the type (a tuple?)

and not tacked onto (~essentially immutable by interface) objects after construction

Reply:
(another alternative)
3. stored in a separate structure

Reply:
Yeah, I agree. But I think we should store it separately somehow since otherwise we will have to change so many places where we create various bindings.

I think this is sound from a type-checking perspective, though? Since `value` is documented to be an instance of `Binding` we know that this property exists.

Reply:
@asottile Actually `Binding` has another attribute `used` that is used in the exact same way as `during_type_checking` currently is. So the more I think about this, the more I think the current way fits into the existing code.

Reply:
Uploaded an example. Not really sure about it, but at least it is another solution.